supabase/migrations/33_automate_trial_start.sql

```diff
@@ -0,0 +1,88 @@
+begin;
+
+create or replace view internal.new_free_trial_tenants as
+with hrs_by_day as (
+  select
+    tenants.tenant as tenant,
+    ts,
+    sum(catalog_stats_daily.usage_seconds / (60.0 * 60)) as daily_usage_hours
+  from catalog_stats_daily
+  join tenants on tenants.tenant = split_part(catalog_stats_daily.catalog_name,'/',1)||'/'
+  where tenants.trial_start is null
+  group by tenants.tenant, ts
+  having sum(catalog_stats_daily.usage_seconds / (60.0 * 60)) > (2 * 24)
+),
+hrs_by_month as (
+  select
+    tenants.tenant as tenant,
+    ts,
+    sum(catalog_stats_monthly.usage_seconds / (60.0 * 60)) as monthly_usage_hours
+  from catalog_stats_monthly
+  join tenants on tenants.tenant = split_part(catalog_stats_monthly.catalog_name,'/',1)||'/'
+  where tenants.trial_start is null
+  group by tenants.tenant, ts
+  having sum(catalog_stats_monthly.usage_seconds / (60.0 * 60)) > (24 * 31 * 2)
+),
+gbs_by_month as (
+  select
+    tenants.tenant as tenant,
+    ts,
+    ceil(sum((catalog_stats_monthly.bytes_written_by_me + catalog_stats_monthly.bytes_read_by_me) / (1024.0 * 1024 * 1024))) as monthly_usage_gbs
+  from catalog_stats_monthly
+  join tenants on tenants.tenant = split_part(catalog_stats_monthly.catalog_name,'/',1)||'/'
+  where tenants.trial_start is null
+  group by tenants.tenant, ts
+  having ceil(sum((catalog_stats_monthly.bytes_written_by_me + catalog_stats_monthly.bytes_read_by_me) / (1024.0 * 1024 * 1024))) > 10
+)
+select
+    tenants.tenant as tenant,
+    max(hrs_by_day.daily_usage_hours) as max_daily_usage_hours,
+    max(hrs_by_month.monthly_usage_hours) as max_monthly_usage_hours,
+    max(gbs_by_month.monthly_usage_gbs) as max_monthly_gb,
+    count(distinct live_specs.id) filter (where live_specs.spec_type = 'capture') as today_captures,
+    count(distinct live_specs.id) filter (where live_specs.spec_type = 'materialization') as today_materializations
+from tenants
+left join hrs_by_day on hrs_by_day.tenant = tenants.tenant
+left join hrs_by_month on hrs_by_month.tenant = tenants.tenant
+left join gbs_by_month on gbs_by_month.tenant = tenants.tenant
+join live_specs on (split_part(live_specs.catalog_name,'/',1)||'/' = tenants.tenant and (live_specs.spec #>> '{shards,disable}')::boolean is not true)
+where tenants.trial_start is null
+group by tenants.tenant
+having (
+  (
```

Comment:
why is this repeated? don't the individual views already have these conditions?

Reply:
This exact check is a bit unclear and leftover from before I pushed the filters into the CTEs, but we do need a filter here to make sure that the tenant actually _has_ at least one non-null criterion pushing them over the free tier. Changed it to `having count(hours_by_month) > 0 or count(hours_by_day) > 0 or count(gbs_by_month) > 0`